src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,50 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final int INDEX_DATE = 0;
+    private static final int INDEX_NAME = 1;
+    private static final int INDEX_WORKING_HOURS = 2;
+    private static final int INDEX_AMOUNT = 3;
+    private static final String SEPARATOR = " ";
+    private static final String DATA_FORMATTER_PATTERN = "dd.MM.yyyy";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder report = new StringBuilder();
+        report.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        report.append(createReports(names, data, dateFrom, dateTo));
+        return report.toString();
```

Comment:
Actually, there is no need to use StringBuilder if you have no loops or other complicated logic as the compiler will optimize your code.